src/com/google/javascript/jscomp/parsing/parser/Parser.java

```diff
@@ -2824,7 +2824,7 @@ private ParseTree parseAssignment(Expression expressionIn) {
 
     if (isStartOfAsyncArrowFunction(left)) {
       // re-evaluate as an async arrow function.
-      resetScanner(left);
+      resetScanner(start);
```

Comment:
If my mental model were correct, then that would mean

`a + b` => AND node source location covers the entire expression, not just the location of the `+`

If that's correct, then the CALL node's source location really must include the entire expression and not point at the `(`.